align_anything/evaluation/inference/base_inference.py

```diff
@@ -0,0 +1,330 @@
+from typing import List, Dict, Any
+from torch.nn.utils.rnn import pad_sequence
+import torch.distributed as dist
+from torch.utils.data import Dataset, DataLoader, DistributedSampler
+import deepspeed
+from vllm.utils import cuda_device_count_stateless
+from transformers.integrations.deepspeed import HfDeepSpeedConfig
+from align_anything.models.pretrained_model import load_pretrained_models
+from align_anything.utils.tools import requestoutput_to_dict
+from align_anything.evaluation.data_type import InferenceInput, InferenceOutput
+from vllm import LLM, SamplingParams
+import pickle
+
+def update_results(output_dir:str,
+                     brief_filename:str,
+                        detailed_filename:str,
+                    task2details:Dict[str, Dict[str, Any]]
+                )->None:
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir, exist_ok=True)
+    brief_file_path = os.path.join(output_dir, brief_filename)
+    detailed_file_path = os.path.join(output_dir, detailed_filename)
+    
+    for task, value in task2details.items():
+        output_brief = []
+        output_detailed = []
+        
+        for item in value:
+            output_brief.append(requestoutput_to_dict(item.raw_output, mode='brief'))
+            output_detailed.append(requestoutput_to_dict(item.raw_output, mode='detailed'))
+            
+        with open(brief_file_path + '_' + task + ".jsonl", 'w', encoding='utf-8') as file:
+            for item in output_brief:
+                json_record = json.dumps(item, ensure_ascii=False)
+                file.write(json_record + '\n')
+
+        with open(detailed_file_path + '_' + task + ".jsonl", 'w', encoding='utf-8') as file:
+            for item in output_detailed:
+                json_record = json.dumps(item, ensure_ascii=False)
+                file.write(json_record + '\n')
+
+class BaseInferencer_vllm:
+    def __init__(self, 
+                 model_cfgs: Dict[str, Any],
+                 vllm_cfgs,
+                 **kwargs):
+        self.vllm_cfgs_sp, self.vllm_cfgs_llm = vllm_cfgs.SamplingParams, vllm_cfgs.LLM
+        self.model_cfgs = model_cfgs
+        self.sp_n = self.vllm_cfgs_sp.n
+        self.sp_top_k = self.vllm_cfgs_sp.top_k
+        self.sp_top_p = self.vllm_cfgs_sp.top_p
+        self.sp_temperature = self.vllm_cfgs_sp.temperature
+        self.sp_max_tokens = self.model_cfgs.model_max_length
+        self.sp_frequency_penalty = self.vllm_cfgs_sp.frequency_penalty
+        self.sp_prompt_logprobs = self.vllm_cfgs_sp.prompt_logprobs
+        self.sp_logprobs = self.vllm_cfgs_sp.logprobs
+
+        self.llm_tokenizer_mode = self.vllm_cfgs_llm.tokenizer_mode
+        self.llm_trust_remote_code = self.vllm_cfgs_llm.trust_remote_code
+        self.llm_gpu_memory_utilization = self.vllm_cfgs_llm.gpu_memory_utilization
+        self.llm_max_num_seqs = self.vllm_cfgs_llm.max_num_seqs
+        self.llm_tensor_parallel_size = cuda_device_count_stateless()
```

Comment:
Try to support command line hyper-parameters, you can also change the config file in vllm_basic.json

I try to implement it by the following command. But it occurs some error : 
```suggestion
        tensor_ps = self.vllm_cfgs_llm.tensor_parallel_size
        self.llm_tensor_parallel_size = tensor_ps if tensor_ps else cuda_device_count_stateless()
```

```
AttributeError: 'RayGPUExecutor' object has no attribute 'forward_dag'
```

Reply:
This problem is caused by the same reason as #35 and will be fixed in the next version.

Reply:
Great! I suggest add some experiments with Vicuna-33B, if it works, then I think it is ok.